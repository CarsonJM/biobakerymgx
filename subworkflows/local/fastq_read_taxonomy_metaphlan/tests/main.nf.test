nextflow_workflow {

    name "Test Subworkflow: FASTQ_READ_TAXONOMY_METAPHLAN"
    script "../main.nf"
    workflow "FASTQ_READ_TAXONOMY_METAPHLAN"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "fastq_read_taxonomy_metaphlan"
    tag "fastq_read_taxonomy_metaphlan_default"


    test(" ['sarscov2']['illumina']['test_fastq_gz'] ") {

        when {
            params {
                outdir                  = "$outputDir"
                metaphlan_db_version    = "mpa_vJan21_TOY_CHOCOPhlAnSGB_202103"
            }
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [ id: 'test1' ],
                        [
                            file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
                            file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true)
                        ]
                    ]
                )
                input[1] = "https://github.com/biobakery/MetaPhlAn/raw/master/metaphlan/utils/mpa_vOct22_CHOCOPhlAnSGB_202212_SGB2GTDB.tsv"
                input[2] = null
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }

    test(" ['sarscov2']['illumina']['test_fastq_gz'] - stub") {

        options "-stub"

        when {
            params {
                outdir                  = "$outputDir"
                metaphlan_db_version    = "mpa_vJan21_TOY_CHOCOPhlAnSGB_202103"
            }
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [ id: 'test' ],
                        [
                            file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
                            file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true)
                        ]
                    ]
                )
                input[1] = "https://github.com/biobakery/MetaPhlAn/raw/master/metaphlan/utils/mpa_vOct22_CHOCOPhlAnSGB_202212_SGB2GTDB.tsv"
                input[2] = null
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }
}
