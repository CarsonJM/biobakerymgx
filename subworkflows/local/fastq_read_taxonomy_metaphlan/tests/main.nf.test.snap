{
    "fastq.gz": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    "versions.yml:md5,4e157c9ddf7bbb9e0f8f8faff2fa6a17",
                    "versions.yml:md5,4e157c9ddf7bbb9e0f8f8faff2fa6a17",
                    "versions.yml:md5,88e7680fe2a8da33e35a81a07afff2c5",
                    "versions.yml:md5,bb024cab3c497b202e7843d975eca0cf",
                    "versions.yml:md5,bb024cab3c497b202e7843d975eca0cf"
                ],
                "metaphlan_profiles_merged_tsv": [
                    
                ],
                "versions": [
                    "versions.yml:md5,4e157c9ddf7bbb9e0f8f8faff2fa6a17",
                    "versions.yml:md5,4e157c9ddf7bbb9e0f8f8faff2fa6a17",
                    "versions.yml:md5,88e7680fe2a8da33e35a81a07afff2c5",
                    "versions.yml:md5,bb024cab3c497b202e7843d975eca0cf",
                    "versions.yml:md5,bb024cab3c497b202e7843d975eca0cf"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-24T10:29:04.201964325"
    }
}